crates/ecs/src/lib.rs

```diff
@@ -861,7 +893,17 @@ impl World {
                 )?;
             }
             None => {
-                self.move_entity_components_to_new_archetype(entity, target_type_ids)?;
+                let mut target_type_ids_vec: Vec<TypeId> =
+                    target_type_ids.clone().into_iter().collect();
```

Comment:
Use `.iter().cloned()` instead of `.clone().into_iter()` to reduce unnecessary allocation. Same thing on line number 716.